File: src/main/kotlin/days/day3/Task1.kt
package days.day3

import framework.Task
import shared.toLines

object Task1 : Task<Int>() {
    override fun run(input: String): Result<Int> {
        val data = input.toLines().toList()

        val count = countTrees(data, 3, 1)

        return Result.success(count)
    }
}

File: 2021/src/tasks/day1/task1.ts
import { Task } from "../../types";

export const task1: Task = (data) => {
  const measurements = data.map((entry) => parseInt(entry, 10));
  let previous,
    count = 0;

  for (const measurement of measurements) {
    if (previous && measurement > previous) {
      count += 1;
    }
    previous = measurement;
  }

  return count;
};
